src/clr-angular/forms/combobox/option.ts

```diff
@@ -3,71 +3,94 @@
+  public optionProxy: OptionProxy<T> = new OptionProxy(null, null);
+
+  private _id: string;
+  @Input('id')
+  set optionId(id: string) {
+    this._id = id;
+    this.optionProxy.id = this._id;
+  }
+  get optionId() {
+    return this._id;
+  }
 
-  @HostBinding('class.active') selected = false;
+  private _value: T;
+  @Input('clrValue')
+  set value(value: T) {
+    this._value = value;
+    this.optionProxy.value = value;
+  }
+  get value(): T {
+    return this._value;
+  }
 
-  @Input('clrValue') value: T;
+  @HostBinding('class.active')
+  get selected() {
+    return (
+      this.optionSelectionService.selectionModel && this.optionSelectionService.selectionModel.containsItem(this.value)
+    );
+  }
 
   constructor(
-    private toggleService: ClrPopoverToggleService,
     @Optional()
     @Inject(POPOVER_HOST_ANCHOR)
     parentHost: ElementRef,
     public elRef: ElementRef,
-    private optionSelectionService: OptionSelectionService<T>
+    private focusHandler: ComboboxFocusHandler<T>,
+    private optionSelectionService: OptionSelectionService<T>,
+    @Inject(UNIQUE_ID) private autoId: string
   ) {
     if (!parentHost) {
       throw new Error('clr-option should only be used inside of a clr-combobox');
     }
-    this.initializeSubscription();
+    this.optionProxy.el = this.elRef.nativeElement;
   }
 
-  private initializeSubscription(): void {
-    this.subscription = this.optionSelectionService.valueChanged.subscribe((value: T) => {
-      // Check for null and undefined needed because if the user doesnt assign a value to the option,
-      // all options should not be selected as the value would be null or undefined
-      if (value === null || value === undefined) {
-        this.selected = false;
-      } else if (this.value === value) {
-        // TODO: Render option when current selection is set by the user
-        this.selected = true;
-      } else {
-        this.selected = false;
-      }
-    });
+  ngOnInit() {
+    if (!this._id) {
+      this._id = 'clr-option-' + this.autoId;
+      this.optionProxy.id = this._id;
+    }
   }
 
   /**
    * This behavior is only for single select. Multi select will keep the menu open on option click.
    * We will handle that later.
    */
   @HostListener('click')
-  updateSelectionAndCloseMenu() {
+  onClick() {
     // We call render here without checking the value because even if the user hasn't
-    // assigned a value to the option, we should atleast display the selection on the input.
+    // assigned a value to the option, we should at least display the selection on the input.
     // This is what the native select does.
-    this.optionSelectionService.renderSelection(this);
-    this.optionSelectionService.updateSelection(this.value);
-    this.toggleService.open = false;
+    this.optionSelectionService.select(this.value);
   }
 
-  // Lifecycle Methods
-  ngOnDestroy(): void {
-    this.subscription.unsubscribe();
+  @HostBinding('class.clr-focus')
```

Comment:
I don't see a test for the true/false states of this hostbinding. Imo it should be part of the `View Basics` suite

Reply:
Thanks! Added these tests.